txpipe/twopoint_null_tests.py

```diff
@@ -0,0 +1,614 @@
+        S.save_fits(self.get_output(sacc_file), overwrite=True)
+
+        # Also make a plot of the data
+
+
+class TXGammaTRandoms(TXTwoPoint):
+    """
+    This subclass of the standard TXTwoPoint uses the centers
+    of stars as "lenses", as a systematics test.
+    """
+
+    name = "TXGammaTRandoms"
+    inputs = [
+        ("calibrated_shear_catalog", ShearCatalog),
+        ("shear_photoz_stack", HDFFile),
+        ("random_cats", RandomsCatalog),
+        ("patch_centers", TextFile),
+        ("tracer_metadata", HDFFile),
+    ]
+    outputs = [
+        ("gammat_randoms", SACCFile),
+        ("gammat_randoms_plot", PNGFile),
+    ]
+    # Add values to the config file that are not previously defined
+    config_options = {
+        "calcs": [0, 1, 2],
+        "min_sep": 2.5,
+        "max_sep": 100,
+        "nbins": 20,
+        "bin_slop": 0.1,
+        "sep_units": "arcmin",
+        "flip_g1": False,
+        "flip_g2": True,
+        "cores_per_task": 20,
+        "verbose": 1,
+        "reduce_randoms_size": 1.0,
+        "var_method": "shot",
+        "npatch": 5,
+        "use_true_shear": False,
+        "subtract_mean_shear": False,
+        "use_randoms": False,
+    }
+
+    def run(self):
+        # Before running the parent class we add source_bins and lens_bins
+        # options that it is expecting, both set to -1 to indicate that we
+        # will choose them automatically (below).
+        import matplotlib
+
+        matplotlib.use("agg")
+        super().run()
+
+    def read_nbin(self):
+        # We use only a single source and lens bin in this case -
+        # the source is the complete 2D field and the lens is the
+        # field centers
+        return ["all"], [0]
+
+    def get_random_catalog(self, i):
+        # override the parent method
+        # so that we don't load the randoms here,
+        # because if we subtract randoms from randoms
+        # we get nothing.
+        return None
+
+    def get_lens_catalog(self, i):
+        import treecorr
+
+        assert i == 0
+        cat = treecorr.Catalog(
+            self.get_input("random_cats"),
+            ext=f"randoms",
+            ra_col="ra",
+            dec_col="dec",
+            ra_units="degree",
+            dec_units="degree",
+            patch_centers=self.get_input("patch_centers"),
+        )
+        return cat
+
+    def select_calculations(self, source_list, lens_list):
+        # We only want a single calculation, the gamma_T around
+        # the field centers
+        return [("all", 0, SHEAR_POS)]
+
+    def write_output(self, source_list, lens_list, meta, results):
+        # we write output both to file for later and to
+        # a plot
+        self.write_output_sacc(meta, results)
+        self.write_output_plot(results)
+
+    def write_output_plot(self, results):
+        import matplotlib.pyplot as plt
+
+        d = results[0]
+        dvalue = d.object.xi
+        derror = np.sqrt(d.object.varxi)
```

Comment:
can we use the full covariance?

Reply:
Opening a new branch to do this - will do a PR soon.